src/core/core.controller.js

```diff
@@ -605,6 +605,11 @@ module.exports = function(Chart) {
 		getDatasetMeta: function(datasetIndex) {
 			var me = this;
 			var dataset = me.data.datasets[datasetIndex];
+			
+			if (typeof dataset == "undefined"){
```

Comment:
Checking `data === 'undefined'` here simply silents the real issue which is: why `getDatasetMeta` is called with an invalid `datasetIndex`? The code that needs to be investigated / changed is the calling method, not this one.